support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -33,28 +34,43 @@ import (
 
 // PodReconcilerConfig holds the config passed in when creating the reconciler
 type PodReconcilerConfig struct {
-	Client             client.Client
-	Cluster            string
-	Ctx                context.Context
-	DisabledNamespaces []string
-	Log                logrus.FieldLogger
-	PodLabel           string
-	PodAnnotation      string
-	Scheme             *runtime.Scheme
-	TrustDomain        string
+	Client               client.Client
+	Cluster              string
+	Ctx                  context.Context
+	DisabledNamespaces   []string
+	Log                  logrus.FieldLogger
+	PodLabel             string
+	PodAnnotation        string
+	Scheme               *runtime.Scheme
+	TrustDomain          string
+	IdentitySchemaConfig string
 }
 
 // PodReconciler holds the runtime configuration and state of this controller
 type PodReconciler struct {
 	client.Client
-	c PodReconcilerConfig
+	c  PodReconcilerConfig
+	is IdentitySchema
 }
 
 // NewPodReconciler creates a new PodReconciler object
 func NewPodReconciler(config PodReconcilerConfig) *PodReconciler {
+
+	idSchema := IdentitySchema{}
+	identitySchemaConfig := config.IdentitySchemaConfig
+	if identitySchemaConfig == "" {
+		identitySchemaConfig = "/run/identity-schema/config/identity-schema.yaml"
+		log.Printf("pod_controller, newPodReconciler: Path to the identity schema config not set. Using default: %s", identitySchemaConfig)
+	}
+	if _, err := idSchema.loadConfig(identitySchemaConfig); err != nil {
```

Comment:
should this be `else if`?

Reply:
I think this is correct. That's a pretty standard syntax. For example:

https://github.com/spiffe/spire/blob/main/support/k8s/k8s-workload-registrar/mode-crd/controllers/node_controller.go#L73

I am refactoring this part a bit here.